proposals/array_interpolate_linear.json

```diff
@@ -0,0 +1,57 @@
+{
+    "id": "array_interpolate_linear",
+    "summary": "One-dimensional linear interpolation for arrays",
+    "description": "Performs a linear interpolation for each of the no-data values (`null`) in the array given, except for the first and last value.",
```

Comment:
> except for the first and last value.

except for leading and trailing no-data values ?

Reply:
Another option is to provide the `fill_value` argument like in `xarray.interpolate_na` or `scipy.interpolate.interp1d`

```python
>>> nan = float("NaN")
>>> data = xarray.DataArray([nan, 1, nan, 2, nan, 10, nan, nan])

>>> data.interpolate_na("dim_0").data
array([ nan,  1. ,  1.5,  2. ,  6. , 10. ,  nan,  nan])

>>> data.interpolate_na("dim_0", fill_value=-1).data
array([-1. ,  1. ,  1.5,  2. ,  6. , 10. , -1. , -1. ])

>>> data.interpolate_na("dim_0", fill_value="extrapolate").data
array([ 0.5,  1. ,  1.5,  2. ,  6. , 10. , 14. , 18. ])
```

Reply:
I had something similar in the first commit (first/last parameters). That seemed to make the process definition much more complex so I decided to start with a minimum and add features for additional requirements once they come up.

Reply:
a problem with this basic spec might be that the user expects that all NaN are (or can be) replaced by `array_interpolate_linear`, regardless of their location. But maybe I'm overthinking it

Reply:
Let's wait for the first implementation / use-case and then try to figure out what to add. :-)